landlab/components/river_bed_dynamics/river_bed_dynamics.py

```diff
@@ -0,0 +1,3433 @@
+                    data = np.hstack(
+                        (
+                            self._current_t,
+                            z[i],
+                            F[i, :],
+                            self._bed_subsurface__grain_size_distribution_link[i, :],
+                        )
+                    )
+                    data = np.reshape(data, [1, data.size])
+                with open("link_" + str(i) + ".txt", "ab") as f:
+                    np.savetxt(f, data, "%.3f")
+                self._bed_surface__grain_size_distribution_link[
+                    i, :
+                ] = self._bed_subsurface__grain_size_distribution_link[i, :]
+
+            self._bed_surface__surface_thickness_new_layer_link[
+                self._id_eroded_links
+            ] = (
+                self._bed_surface__surface_thickness_new_layer_link[
+                    self._id_eroded_links
+                ]
+                + self._new_surface_layer_thickness
+            )
+            self._topographic__elevation_subsurface_link[self._id_eroded_links] = (
+                self._topographic__elevation_subsurface_link[self._id_eroded_links]
+                - self._new_surface_layer_thickness
+            )
+
+            self._subsurface_changed = True
+
+        os.chdir(self._cwd)
+        self._stratigraphy_cycle = 0
+        self._compute_stratigraphy = False
+        self._update_subsurface = False
+        self._update_subsurface_eroded = False
+
+    def bedload_mapper(self):
+        """Maps the magnitude of the volumetric bedload per unit width values
+        from links (m2/s) onto nodes (m3/s).
+
+        This method takes the bedload transport rates from links and calculates
+        the magnitude at a given node.
+        """
+
+        qb_x_r = (
+            self._sediment_transport__bedload_rate_link[self._grid.links_at_node[:, 0]]
+            * self._grid.dy
+        )
+        (I,) = np.where(self._grid.links_at_node[:, 0] < 0)
+        qb_x_r[I] = 0
+
+        qb_x_l = (
+            self._sediment_transport__bedload_rate_link[self._grid.links_at_node[:, 2]]
+            * self._grid.dy
+        )
+        (I,) = np.where(self._grid.links_at_node[:, 2] < 0)
+        qb_x_l[I] = 0
+
+        qb_y_u = (
+            self._sediment_transport__bedload_rate_link[self._grid.links_at_node[:, 1]]
+            * self._grid.dx
+        )
+        (I,) = np.where(self._grid.links_at_node[:, 1] < 0)
+        qb_y_u[I] = 0
+
+        qb_y_l = (
+            self._sediment_transport__bedload_rate_link[self._grid.links_at_node[:, 3]]
+            * self._grid.dx
+        )
+        (I,) = np.where(self._grid.links_at_node[:, 3] < 0)
+        qb_y_l[I] = 0
+
+        qb_x = 0.5 * (qb_x_r + qb_x_l)
+        qb_y = 0.5 * (qb_y_u + qb_y_l)
+        print
+
+        self._bedloadRate_vector = np.transpose(np.vstack((qb_x, qb_y)))
+        self._bedloadRate_magnitude = np.sqrt(qb_x**2 + qb_y**2)
+
+    def vector_mapper(self, vector):
```

Comment:
This seems like an example of a method that could really be a separate function. That is,
```python
def vector_mapper(grid, vector):
    ...
```

Reply:
Thanks, done!